src/utils_internal.rs

```diff
@@ -270,10 +270,10 @@ where
 ///
 pub fn format_data_packet(packet: EncodedToRadioPacket) -> EncodedToRadioPacketWithHeader {
     let data = packet.data();
```

Comment:
This function fails to encode packets that are longer than 2 bytes, I suggest changing the signature.
```suggestion
pub fn format_data_packet(packet: EncodedToRadioPacket) -> Result<EncodedToRadioPacketWithHeader, Error> {
    let data = packet.data();
    if data.len() >= (1 << 16) {
        return Err(Error ....);
    }
```

Reply:
Good call, I'm open to the signature change! Would that affect the signature of any function that calls `format_data_packet`?

Reply:
Not necessarily, if the caller already returns `Result<...>`, you'd just add `?`. E.g. `let data_with_header = format_data_packet(..)?;` inside `stream_api.send_raw`.

Actually, `send_raw` is the only caller of `format_data_packet`, so the only change outside of this function is a single added `?`.

Reply:
Perfect! Happy to push a PR for that signature change and merge this now, also happy to include the signature change in this PR. Your call 🙂

Reply:
I ended up doing it in this PR, it's relevant after all. The big change is the addition of  `Error::InvalidaDataSize`. Please check if you are fine with the name.

Reply:
Yeah no problems with the name!